scripts/xdcc_autoget.pl

```diff
@@ -0,0 +1,711 @@
+	Irssi::print "ag_xdcc_cancel_prefix    : the xdcc message to cancel a transfer";
+	Irssi::print "ag_xdcc_find_prefix      : the xdcc message before the search term";
+	Irssi::print "ag_format                : universal string appended to the end of each search in episodic. Use if more than one format exists";
+	Irssi::print "ag_folder                : Location for data files. ~/.irssi/ reccomended";
+}
+
+sub ag_server	#sets the current server
+{
+	$server = Irssi::active_server();	#keep trying to get server until it works
+	if ($server !~ m/^Irssi::Irc::Server=HASH/) {Irssi::timeout_add_once(1000, sub {&ag_server;} , []);}	
+}
+
+sub ag_getbots		#reads in bot list
+{
+	open(bots, "<", $botsfilename);
+	@bots = <bots>;
+	chomp(@bots);
+	close(bots);
+}
+
+sub ag_getterms		#reads in search term list
+{
+	open(SEARCHES, "<", $searchesfilename);
+	@terms = <SEARCHES>;
+	chomp(@terms);
+	close(SEARCHES);
+}
+
+sub ag_getfinished		#reads in finished packs list
+{
+	open(FINISHED, "<", $cachefilename);
+	@finished = <FINISHED>;
+	chomp(@finished);
+	@finished = ag_uniq(@finished);
+	close(FINISHED);
+}
+
+sub ag_clearcache		#clears cache of saved packs
+{
+	unlink $cachefilename;
+	open(FINISHED, ">>", $cachefilename);
+	close(FINISHED);
+}
+
+sub ag_search		#searches bots for packs
+{
+ 	$msgflag = 0;	#unset message flag so that ag_skip knows no important message has arrived
+	if($episodicflag)	#episodic searches are complicated
+	{
+		my $ep = sprintf("%.2d", $episode);
+		if ($format ne "" and $formatflag)
+		{
+			&ag_message("msg $bots[$botcounter] $findprefix $terms[$termcounter] $ep $format");	#first search with format
+			if ($episode == 1)
+			{
+				push(@totags, Irssi::timeout_add_once($botdelay * 1000, sub { $formatflag = 0; &ag_search; } , []));	#if no formated version found, try again with just the search string + ep#
+			}
+			else
+			{
+				push(@totags, Irssi::timeout_add_once($botdelay * 1000, sub { &ag_skip; } , []));	
+			}
+		}
+		else
+		{
+			&ag_message("msg $bots[$botcounter] $findprefix $terms[$termcounter] $ep" );
+			push(@totags, Irssi::timeout_add_once($botdelay * 1000, sub { &ag_skip; } , []));
+		}
+	}
+	else		#if not episodic, just search and skip
+	{
+		&ag_message("msg $bots[$botcounter] $findprefix $terms[$termcounter]" );
+		push(@totags, Irssi::timeout_add_once($botdelay * 1000, sub { &ag_skip; } , []));
+	}
+	Irssi::signal_add("message irc notice", "ag_getmsg");
+}
+
+sub ag_remtimeouts	#remove timeouts to avoid multiple instances of everything
+{
+	foreach my $to (@totags)	#remove timeouts and clear array
+	{
+		Irssi::timeout_remove($to);
+	}
+	@totags = ();
+}
+
+sub ag_getmsg		#runs when bot sends privmsg. Avoid talking to bots to keep this from sending useless shit that breaks things
+{	
+	my $message = $_[1];
+	my $botname = $_[2];
+	$botname =~ tr/[A-Z]/[a-z]/;
+	$bots[$botcounter] =~ tr/[A-Z]/[a-z]/;
+	
+	if ($botname == $bots[$botcounter] and !$msgflag)	#if it's your bot
```

Comment:
this cannot work, in Perl strings(text) must be compared with `eq`. `==` will only compare numbers (or turn text into numbers first)